storage/filesystem/internal/dotgit/dotgit_test.go

```diff
@@ -216,7 +260,92 @@ func (s *SuiteDotGit) TestPackfile(c *C) {
 	}
 }
 
+func (s *SuiteDotGit) TestObjectfiles(c *C) {
+	objectfiles := func(d *DotGit) (fs.FS, []core.Hash, error) {
```

Comment:
I don't understand the point of this variable `objectfiles`.

Isn't the purpose of this method to test `DotGit.Objectfiles`?, then why are you wrapping it with the `objectfiles` variable?

Just call `DotGit.Objectfiles` directly on the tests, otherwise it is too convoluted.